### PR TITLE
add virt/virtualization

### PR DESCRIPTION
**Pull request by svenfuchs**

### CODE REVIEW

File: lib/travis/yml/schema/def/virt.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          def define
+            title 'Virtualization'
+            summary 'Build environment virtualization'
+            # see 'Virtualization environments': 'https://docs.travis-ci.com/user/reference/overview/#virtualization-environments'
+
+            aliases :virtualization
+
+            downcase
+
+            supports only: { arch: 'ppc64le'}
```
Comment (michal-at-travisci) [inline lib/travis/yml/schema/def/virt.rb]: also `amd64` as a second available item

File: lib/travis/yml/schema/def/virt.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            default :lxd, only: { arch: 'ppc64le' }
```
Comment (michal-at-travisci) [inline lib/travis/yml/schema/def/virt.rb]: also `amd64` as a second available item

File: schema.json [hunk 3]
```diff
@@ -1948,6 +1951,35 @@
...
+        "title": "Virtualization",
+        "type": "string",
+        "summary": "Build environment virtualization",
+        "aliases": [
+          "virtualization"
+        ],
+        "downcase": true,
+        "only": {
+          "arch": [
+            "ppc64le"
```
Comment (michal-at-travisci) [inline schema.json]: also `amd64` as a second available item

File: schema.json [hunk 4]
```diff
@@ -1948,6 +1951,35 @@
...
+          ]
+        },
+        "defaults": [
+          {
+            "value": "lxd",
+            "only": {
+              "arch": [
+                "ppc64le"
```
Comment (michal-at-travisci) [inline schema.json]: default should be set to `amd64`

File: spec/travis/yml/accept/virt_spec.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    it { should serialize_to empty }
+  end
+
+  known = %w(
+    lxd
+    vm
+  )
+
+  known.each do |value|
+    describe "given #{value}" do
```
Comment (michal-at-travisci) [inline spec/travis/yml/accept/virt_spec.rb]: similar rule should be for:
arch: amd64
virt: #{value}

File: spec/travis/yml/accept/virt_spec.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  describe 'given an unknown value (no arch)' do
+    yaml %(
+      virt: unknown
+    )
+    it { should serialize_to virt: 'lxd' }
+    it { should have_msg [:warn, :virt, :unknown_default, value: 'unknown', default: 'lxd'] }
+  end
+
+  describe 'given an unknown value (arch ppc64le)' do
```
Comment (michal-at-travisci) [inline spec/travis/yml/accept/virt_spec.rb]: `arch: amd64` with unknown virt - it should also default to `lxd`

File: spec/travis/yml/schema/def/virt_spec.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      type: :string,
+      aliases: [
+        :virtualization
+      ],
+      enum: [
+        'lxd',
+        'vm'
+      ],
+      downcase: true,
+      only: {
```
Comment (michal-at-travisci) [inline spec/travis/yml/schema/def/virt_spec.rb]: `amd64` in the same list?

File: spec/travis/yml/accept/virt_spec.rb (hunk 5 above)
Comment (svenfuchs) [reply spec/travis/yml/accept/virt_spec.rb]: removed that for now, @michal-at-travisci

**Pull request merged**